ste/sender-blobSymlinks.go

```diff
@@ -68,16 +68,33 @@ func (s *blobSymlinkSender) SendSymlink(linkData string) error {
 	}
 	s.metadataToApply["is_symlink"] = to.Ptr("true")
 
+	blobTags := s.blobTagsToApply
+	setTags := separateSetTagsRequired(blobTags)
+	if setTags || len(blobTags) == 0 {
+		blobTags = nil
+	}
+
 	_, err = s.destinationClient.Upload(s.jptm.Context(), streaming.NopCloser(strings.NewReader(linkData)),
 		&blockblob.UploadOptions{
 			HTTPHeaders:  &s.headersToApply,
 			Metadata:     s.metadataToApply,
 			Tier:         s.destBlobTier,
-			Tags:         s.blobTagsToApply,
+			Tags:         blobTags,
 			CPKInfo:      s.jptm.CpkInfo(),
 			CPKScopeInfo: s.jptm.CpkScopeInfo(),
 		})
-	return err
+	if err != nil {
```

Comment:
was this a bug - us not doing a separate set tags in certain situations for symlinks? Should we add an explicit symlink test?

Reply:
Yes, it appears so. I'll port this to folders and modify the tests.